test/filterResults.test.ts

```diff
@@ -0,0 +1,79 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import chaiString from 'chai-string'
+import fs from 'fs/promises'
+import os from 'os'
+import path from 'path'
+import spawn from 'spawn-please'
+import { FilterResultsFunction } from '../src/types/FilterResultsFunction'
+import stubNpmView from './helpers/stubNpmView'
+
+chai.should()
+chai.use(chaiAsPromised)
+chai.use(chaiString)
+
+const bin = path.join(__dirname, '../build/src/bin/cli.js')
+
+/**
+ * Sets up and tears down the temporary directories required to run each test
+ */
+async function filterResultsTestScaffold(
+  dependencies: Record<string, string>,
+  filterResultsFn: FilterResultsFunction,
+  expectedOutput: string,
+  notExpectedOutput: string,
+): Promise<void> {
+  const stub = stubNpmView(
+    {
+      'ncu-test-v2': '3.0.0',
+      'ncu-test-tag': '2.1.0',
+      'ncu-test-return-version': '1.2.0',
+    },
+    { spawn: true },
+  )
+
+  // use dynamic import for ESM module
+  const { default: stripAnsi } = await import('strip-ansi')
```

Comment:
If you add `jsonUpgraded` to the options, you can just `JSON.parse` the output and avoid having to strip ansi.